### ISSUE TITLE
Loading partials does not work

### ISSUE THREAD

**Issue by TheMaverickProgrammer** [jmjuanes/electron-ejs#8]
I am loading the relative path in the app this way 

```
// load the app first page
const url = `file://${__dirname}/../views/pages/mapping.ejs`
win.loadURL(url)
```

Without partials it works fine, with partials in the EJS script, `File Not Found` exceptions are thrown and bubble up the app. I have tried passing the relative directory into the EJS scripts and loading the partials that way with no luck.

**Comment by jmjuanes** [comment]
Hello! 
I am not able to reproduce this error. I use this module and it works as expected. Please, can you give me the main `.js` file and the `.ejs` files where you get the error?

**Comment by gamelaster** [comment]
Hey,
Same situation, I cant reproduce it!

**Comment by TheMaverickProgrammer** [comment]
Here's my window creation code in the app:

```
const {app, BrowserWindow} = require('electron')
const menu   = require('./menu')
const config  = require('../config')
const ejs        = require('electron-ejs');

// var ejsParser = new ejs();

// Keep a global reference of the window object, if you don't, the window will
// be closed automatically when the JavaScript object is garbage collected.

let win

var createWindow = function() {
  // Create the browser window.
  win = main.win = new BrowserWindow({width: 800, height: 600,
                                      darkTheme: true, frame: true,
                                      title: config.APP_WINDOW_TITLE,
                                      resizable: false})

  // and load the index.html of the app.
  const url = config.PAGES.MAPPING;
  console.log(url)
  win.loadURL(url)
```

where the url is `'file://' + path.join(__dirname, 'views', 'pages', 'mapping.ejs')`

which prints `file:///Users/maverick/Code/Lit Up Games/boomsheets/boomsheets-desktop/views/pages/mapping.ejs`

and loads! So the file is located just fine. However, uncomment the creation of the `ejsParser` object. 

Run the program and `File Note Found` exceptions are thrown from the app. Nothing renders.

```
electron-ejs/index.js:59:16
pstat/index.js:14:34
pstat/index.js:68:25
```

Here's the EJS code for `mapping.ejs`:

```
<% include ../partials/header %>

<body>
    <% include ../partials/navbar %>
  <% include ../partials/sidebar %>

    <div class="col-sm-9 col-sm-offset-3 col-lg-10 col-lg-offset-2 main">
        <div class="row">
            <ol class="breadcrumb">
                <li><a href="#"><span class="glyphter icon-speedometer"></span></a></li>
                <li class="active">Map Frames</li>
            </ol>
            <div class="col-md-8">
                <h1 class="page-header">Mapping<a target="_blank" href="#" title="Click to add frames for the currently selected animation"><img src="images/blue-question-mark.png" height="13px"/></a></h1>
                <div class="row">
                    <div class="col-md-6">
                        <button id="btnScaleUp" title="Scale the image up by 50%"><span class="glyphter icon-zoom-plus tool-icon"></span></button>
                        <button id="btnScaleDown" title="Scale the image down by 25%"><span class="glyphter icon-zoom-minus tool-icon"></span></button>
                        <button id="btnUndo" title="Undo the last action"><span class="glyphter icon-undo tool-icon"></span> Undo</button>
                        <button id="btnRedo" title="Redo the last action"><span class="glyphter icon-undo tool-icon"></span> Redo</button>
                    </div>
                </div>
                </br>
```

where the folders `pages` and `partials` share the same parent folder. So if the mapping.ejs file is loaded from pages, the partials need to go up a directory and into the partials directory.

For an extra check, I deleted the ejs code for including partials so that it is 100% HTML. Same issue! Seems to be with the `ejsParser` object.

**Comment by jmjuanes** [comment]
Hello @TheMaverickProgrammer, sorry for the delay.

I have tried it simulating the same folder structure and with the same app script, and it works...

![image](https://cloud.githubusercontent.com/assets/5751201/19015139/af68ac86-87fe-11e6-8625-6131bf44237d.png)

![image](https://cloud.githubusercontent.com/assets/5751201/19015146/dab7e5b4-87fe-11e6-8cbb-32f49a2f933d.png)

app.js file: 

``` javascript
//Import dependencies
var electron = require('electron');
var electronEjs = require('electron-ejs');
var path = require('path');

//Import electron modules
var ipcMain = electron.ipcMain; //Module for channel messages
var app = electron.app; //Module to control application life.
var BrowserWindow = electron.BrowserWindow; //Module to create native browser window.
var shell = electron.shell; //For desktop integration

//Initialize ejs compiler
var ejs = new electronEjs();

var win = null;

//Initialize the application
app.on('ready', function()
{
  win = new BrowserWindow({width: 800, height: 600,
                                      darkTheme: true, frame: true,
                                      title: 'Test',
                                      resizable: false})

  var url = 'file://' + path.join(__dirname, 'views', 'pages', 'mapping.ejs');
  console.log(url);
  win.loadURL(url);
});

//Kill the app
app.on('window-all-closed', app.quit);
```

mapping.ejs file 

```
<% include ../partials/header %>
<% include ../partials/navbar %>
<% include ../partials/sidebar %>

mapping.ejs file<br>
```

header.ejs file 

```
header.ejs file<br>
```

I don't know why you are getting this error. There is any way that I can see your original code to find the bug?

**Comment by TheMaverickProgrammer** [comment]
Man this is strange. I can temporarily add you to the project on bitbucket. Sent you an invite. Check your gmail. We'll update this issue here when we resolve it.

**Comment by TheMaverickProgrammer** [comment]
Resolved the issue. There were spaces in some of my computer's directories. Created a fix and sent a pull request.

**Comment by BharamuBanajwad** [comment]
Hi @TheMaverickProgrammer, facing the same issue, could you please elaborate how did you fix it?

**Comment by TheMaverickProgrammer** [comment]
@BharamuBanajwad my fix is available since v1.1.1 See the merge here:

https://github.com/jmjuanes/electron-ejs/pull/9

Simply put, resource paths have spaces in them and the url reader was not written to handle that.

**Comment by BharamuBanajwad** [comment]
Hey @TheMaverickProgrammer, 
Found the solution.
Space wasn't the issue in my case in fact didn't had spaces in my paths. Problem is, all the specified links(css) and scripts(js) files should be present in the paths specified otherwise it will halt the entire process.

Thank you.
